pkg/operator/certrotation/target_test.go

```diff
@@ -433,7 +437,83 @@ func TestEnsureTargetSignerCertKeyPair(t *testing.T) {
 				t.Errorf("missing %q", test.expectedError)
 			}
 
-			test.verifyActions(t, client)
+			test.verifyActions(t, client, newCA)
+		})
+	}
+}
+
+func TestSignerSignatureRotation(t *testing.T) {
+
+	tests := []struct {
+		name string
+		caFn func() (*crypto.CA, error)
+	}{
+		{
+			name: "no authority set", // this goes through the more expensive signing check path
+			caFn: func() (*crypto.CA, error) {
+				return newTestCACertificate(pkix.Name{CommonName: "signer-tests"}, int64(1), metav1.Duration{Duration: time.Hour * 24 * 60}, time.Now)
+			},
+		},
+		{
+			name: "authority set", // cheap check against authority key
+			caFn: func() (*crypto.CA, error) {
+				ski := make([]byte, 32)
+				_, err := cryptorand.Read(ski)
+				if err != nil {
+					return nil, err
+				}
+
+				return newTestCACertificateWithAuthority(pkix.Name{CommonName: "signer-tests"}, int64(1), metav1.Duration{Duration: time.Hour * 24 * 60}, time.Now, ski)
+			},
+		},
+	}
```

Comment:
I'd expect 4 tests, splitting each of these in two:
1. signer matches the leaf in the secret
2. signer does not match

You may want to add Reactors to the mocked clients, checking for Updates